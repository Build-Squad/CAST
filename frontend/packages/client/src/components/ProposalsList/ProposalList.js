import React from "react";
import { Link } from "react-router-dom";
import DropDownFilter from "./DropdownFilter";
import ProposalHeader from "./ProposalHeader";
import Loader from "../Loader";

// Used for adding margin on mobile to children
const PaddingWrapper = ({ children }) => {
  return (
    <>
      <div className="is-hidden-mobile column p-0">{children}</div>
      <div className="is-hidden-tablet column p-0 mt-2 mb-5">{children}</div>
    </>
  );
};

const ProposalList = ({
  proposalsList,
  proposalFilterValues,
  setFilterValues = () => {},
  filterValue,
  initialLoading,
} = {}) => {
  return (
    <>
      <div className="columns m-0">
        <div className="column p-0 is-10-tablet is-hidden-mobile">
          <h1 className="title is-4 mb-6" style={{ lineHeight: "1.67" }}>
            Proposals
          </h1>
        </div>
        <PaddingWrapper>
          <DropDownFilter
            value={filterValue}
            filterValues={proposalFilterValues}
            setFilterValues={setFilterValues}
            className="is-hidden-tablet"
          />
        </PaddingWrapper>
      </div>
      {initialLoading && <Loader fullHeight />}
      <div className="is-flex is-flex-direction-column">
        {(proposalsList ?? []).map((pr, i) => (
          <Link to={`/proposal/${pr.id}`} key={i} data-testid="proposal-card">
            <div
              className="border-light rounded-sm mb-5 proposal-card transition-all"
              key={i}
            >
              <ProposalHeader {...pr} />
              <div className="px-6 py-5 proposal-body-spacing">
                <h4 className="proposal-title is-4 mt-1 mb-2">{pr.name}</h4>
                <p className="has-text-grey mb-4 small-text">
                  by {pr.creatorAddr}
                </p>
                <p className="has-text-grey small-text">{pr.description}</p>
              </div>
            </div>
          </Link>
        ))}
      </div>
    </>
  );
};

export default ProposalList;
