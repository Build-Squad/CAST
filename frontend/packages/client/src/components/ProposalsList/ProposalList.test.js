import React from "react";
import { cleanup, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import ProposalList from "./ProposalList";
import { MemoryRouter } from "react-router-dom";

jest.mock("../../utils.js", () => {
  const originalModule = jest.requireActual("../../utils.js");

  return {
    __esModule: true,
    ...originalModule,
    parseDateFromServer: () => ({
      date: new Date(1648726329708),
      diffFromNow: 86400000,
      diffDays: 1,
    }),
  };
});

const date = new Date();

const getProposalList = ({ isFixed = true, filter = "All" } = {}) =>
  [
    {
      id: 1,
      name: "test proposal",
      description: "test description",
      communityId: 1,
      choices: ["a", "b", "c"],
      strategy: "token-weighted-default",
      creatorAddr: "0xf8d6e0586b0a20c7",
      ...(isFixed
        ? {
            startTime: "2022-03-24T11:32:09.708727Z",
            endTime: "2022-03-25T11:32:09.708727Z",
          }
        : {
            startTime: new Date().setDate(date.getDay() - 1),
            endTime: new Date().setDate(date.getDay() + 3),
          }),
      createdAt: "2022-03-24T14:32:09.708727Z",
      status: "published",
      block_height: 0,
      total_votes: 0,
      timestamp: "",
      sig: "",
      computedStatus: "active",
      winCount: 4480000,
      textDecision: "No, do not compensate them",
      voted: false,
    },
    {
      id: 2,
      name: "test proposal",
      description: "test description",
      communityId: 1,
      choices: ["a", "b", "c"],
      strategy: "token-weighted-default",
      creatorAddr: "0xf8d6e0586b0a20c7",
      ...(isFixed
        ? {
            startTime: "2022-03-31T11:32:09.708727Z",
            endTime: "2022-04-23T11:32:09.708727Z",
          }
        : {
            startTime: new Date().setDate(date.getDay() + 1),
            endTime: new Date().setDate(date.getDay() + 3),
          }),
      createdAt: "2022-03-24T14:32:09.708727Z",
      status: "published",
      block_height: 0,
      total_votes: 0,
      timestamp: "",
      sig: "",
      computedStatus: "pending",
      winCount: 4480000,
      textDecision: "No, do not compensate them",
      voted: false,
    },
    {
      id: 3,
      name: "test proposal",
      description: "test description",
      communityId: 1,
      choices: ["a", "b", "c"],
      strategy: "token-weighted-default",
      creatorAddr: "0xf8d6e0586b0a20c7",

      ...(isFixed
        ? {
            startTime: "2022-02-22T11:32:09.708727Z",
            endTime: "2022-03-23T11:32:09.708727Z",
          }
        : {
            startTime: new Date().setDate(date.getDay() - 3),
            endTime: new Date().setDate(date.getDay() - 1),
          }),
      createdAt: "2022-03-24T14:32:09.708727Z",
      status: "published",
      block_height: 0,
      total_votes: 0,
      timestamp: "",
      sig: "",
      computedStatus: "closed",
      winCount: 4480000,
      textDecision: "No, do not compensate them",
      voted: false,
    },
    {
      id: 4,
      name: "test proposal",
      description: "test description",
      communityId: 1,
      choices: ["a", "b", "c"],
      strategy: "token-weighted-default",
      creatorAddr: "0xf8d6e0586b0a20c7",
      createdAt: "2022-03-24T14:32:09.708727Z",
      ...(isFixed
        ? {
            startTime: "2022-03-24T11:32:09.708727Z",
            endTime: "2022-03-25T11:32:09.708727Z",
          }
        : {
            startTime: new Date().setDate(date.getDay() - 3),
            endTime: new Date().setDate(date.getDay() + 1),
          }),
      status: "cancelled",
      block_height: 0,
      total_votes: 0,
      timestamp: "",
      sig: "",
      computedStatus: "cancelled",
      winCount: 4480000,
      textDecision: "No, do not compensate them",
      voted: false,
    },
  ].filter(
    (proposal) =>
      proposal.computedStatus === filter.toLocaleLowerCase() || filter === "All"
  );

const filterValues = ["All", "Active", "Pending", "Closed", "Cancelled"];

afterEach(cleanup);

beforeAll(() => {
  window.matchMedia = (query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  });
});
describe("ProposalList UI Component", () => {
  const testCases = [
    {
      description: "with regular params",
      params: {
        proposalList: getProposalList(),
        filterValues,
        filterValue: filterValues[0],
      },
    },
    {
      description: "with empty proposal list",
      params: { proposalList: [], filterValues, filterValue: filterValues[0] },
    },
    {
      description: "with no params",
      params: {
        proposalList: undefined,
        filterValues: undefined,
        filterValue: undefined,
      },
    },
  ];
  testCases.forEach((testCase) => {
    it(`renders correctly ${testCase.description}`, () => {
      const setFilterValuesMockFn = jest.fn();
      const component = renderer.create(
        <MemoryRouter>
          <ProposalList
            proposalsList={testCase.params.proposalList}
            proposalFilterValues={testCase.params.filterValues}
            setFilterValues={undefined}
            filterValue={testCase.params.filterValue}
          />
        </MemoryRouter>
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});

describe("ProposalList no filter applied (All)", () => {
  let rendered;
  beforeEach(() => {
    const setFilterValuesMockFn = jest.fn();
    rendered = render(
      <MemoryRouter>
        <ProposalList
          proposalsList={getProposalList({ isFixed: false })}
          proposalFilterValues={filterValues}
          setFilterValues={setFilterValuesMockFn}
          filterValue={filterValues[0]}
        />
      </MemoryRouter>
    );
  });
  it("should display 'All' in filter dropdown", async () => {
    const { queryAllByTestId } = rendered;
    // 1 for mobile and one for other sizes
    expect(queryAllByTestId("drop-down-item-All")).toHaveLength(2);
    queryAllByTestId("drop-down-item-All").forEach((element) => {
      expect(element).toHaveClass("is-active");
    });
  });
  it("should display all proposals", () => {
    const { queryAllByTestId } = rendered;
    expect(queryAllByTestId("proposal-card")).toHaveLength(4);
  });
});

describe("ProposalList 'active' status", () => {
  let rendered;
  beforeEach(() => {
    const setFilterValuesMockFn = jest.fn();
    rendered = render(
      <MemoryRouter>
        <ProposalList
          proposalsList={getProposalList({
            isFixed: false,
            filter: filterValues[1],
          })}
          proposalFilterValues={filterValues}
          setFilterValues={setFilterValuesMockFn}
          filterValue={filterValues[1]}
        />
      </MemoryRouter>
    );
  });
  it("should display 'Active' in filter dropdown", async () => {
    const { queryAllByTestId } = rendered;
    // 1 for mobile and one for other sizes
    expect(queryAllByTestId("drop-down-item-Active")).toHaveLength(2);
    queryAllByTestId("drop-down-item-Active").forEach((element) => {
      expect(element).toHaveClass("is-active");
    });
  });
  it("should display active proposals", () => {
    const { queryAllByTestId } = rendered;
    expect(queryAllByTestId("proposal-card")).toHaveLength(1);
  });
});

describe("ProposalList 'pending' status", () => {
  let rendered;
  beforeEach(() => {
    const setFilterValuesMockFn = jest.fn();
    rendered = render(
      <MemoryRouter>
        <ProposalList
          proposalsList={getProposalList({
            isFixed: false,
            filter: filterValues[2],
          })}
          proposalFilterValues={filterValues}
          setFilterValues={setFilterValuesMockFn}
          filterValue={filterValues[2]}
        />
      </MemoryRouter>
    );
  });
  it("should display 'Pending' in filter dropdown", async () => {
    const { queryAllByTestId } = rendered;
    // 1 for mobile and one for other sizes
    expect(queryAllByTestId("drop-down-item-Pending")).toHaveLength(2);
    queryAllByTestId("drop-down-item-Pending").forEach((element) => {
      expect(element).toHaveClass("is-active");
    });
  });
  it("should display pending proposals", () => {
    const { queryAllByTestId } = rendered;
    expect(queryAllByTestId("proposal-card")).toHaveLength(1);
  });
});

describe("ProposalList 'closed' status", () => {
  let rendered;
  beforeEach(() => {
    const setFilterValuesMockFn = jest.fn();
    rendered = render(
      <MemoryRouter>
        <ProposalList
          proposalsList={getProposalList({
            isFixed: false,
            filter: filterValues[3],
          })}
          proposalFilterValues={filterValues}
          setFilterValues={setFilterValuesMockFn}
          filterValue={filterValues[3]}
        />
      </MemoryRouter>
    );
  });
  it("should display 'Closed' in filter dropdown", async () => {
    const { queryAllByTestId } = rendered;
    // 1 for mobile and one for other sizes
    expect(queryAllByTestId("drop-down-item-Closed")).toHaveLength(2);
    queryAllByTestId("drop-down-item-Closed").forEach((element) => {
      expect(element).toHaveClass("is-active");
    });
  });
  it("should display closed proposals", () => {
    const { queryAllByTestId } = rendered;
    expect(queryAllByTestId("proposal-card")).toHaveLength(1);
  });
});

describe("ProposalList 'cancelled' status", () => {
  let rendered;
  beforeEach(() => {
    const setFilterValuesMockFn = jest.fn();
    rendered = render(
      <MemoryRouter>
        <ProposalList
          proposalsList={getProposalList({
            isFixed: false,
            filter: filterValues[4],
          })}
          proposalFilterValues={filterValues}
          setFilterValues={setFilterValuesMockFn}
          filterValue={filterValues[4]}
        />
      </MemoryRouter>
    );
  });
  it("should display 'Cancelled' in filter dropdown", async () => {
    const { queryAllByTestId } = rendered;
    // 1 for mobile and one for other sizes
    expect(queryAllByTestId("drop-down-item-Cancelled")).toHaveLength(2);
    queryAllByTestId("drop-down-item-Cancelled").forEach((element) => {
      expect(element).toHaveClass("is-active");
    });
  });
  it("should display cancelled proposals", () => {
    const { queryAllByTestId } = rendered;
    expect(queryAllByTestId("proposal-card")).toHaveLength(1);
  });
});
