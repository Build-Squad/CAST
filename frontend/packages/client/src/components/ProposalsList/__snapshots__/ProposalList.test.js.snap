// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProposalList UI Component renders correctly with empty proposal list 1`] = `
Array [
  <div
    className="columns m-0"
  >
    <div
      className="column p-0 is-10-tablet is-hidden-mobile"
    >
      <h1
        className="title is-4 mb-6"
        style={
          Object {
            "lineHeight": "1.67",
          }
        }
      >
        Proposals
      </h1>
    </div>
    <div
      className="is-hidden-mobile column p-0"
    >
      <div
        aria-controls="dropdown-menu"
        aria-haspopup="true"
        className="dropdown is-right is-flex is-flex-grow-1"
        onBlur={[Function]}
      >
        <div
          className="dropdown-trigger columns m-0 is-flex-grow-1"
        >
          <button
            aria-controls="dropdown-menu"
            aria-haspopup="true"
            className="button rounded-sm is-outlined border-light column m-0 py-0 px-3 is-full-mobile"
            onClick={[Function]}
          >
            <span
              className="has-tooltip-arrow has-tooltip-top is-flex is-flex-grow-1"
              data-tooltip="Filter proposals based on status"
            >
              <div
                className="is-flex is-flex-grow-1 is-align-items-center is-justify-content-space-between has-text-grey small-text"
              >
                All
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-caret-down has-text-black"
                  data-icon="caret-down"
                  data-prefix="fas"
                  focusable="false"
                  height="15"
                  role="img"
                  viewBox="0 0 320 512"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M310.6 246.6l-127.1 128C176.4 380.9 168.2 384 160 384s-16.38-3.125-22.63-9.375l-127.1-128C.2244 237.5-2.516 223.7 2.438 211.8S19.07 192 32 192h255.1c12.94 0 24.62 7.781 29.58 19.75S319.8 237.5 310.6 246.6z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </span>
          </button>
        </div>
        <div
          className="dropdown-menu column p-0 is-full-mobile"
          id="dropdown-menu"
          role="menu"
        >
          <div
            className="dropdown-content"
          >
            <button
              className="button is-white dropdown-item has-text-grey is-active"
              data-testid="drop-down-item-All"
              onMouseDown={[Function]}
            >
              All
            </button>
            <button
              className="button is-white dropdown-item has-text-grey"
              data-testid="drop-down-item-Active"
              onMouseDown={[Function]}
            >
              Active
            </button>
            <button
              className="button is-white dropdown-item has-text-grey"
              data-testid="drop-down-item-Pending"
              onMouseDown={[Function]}
            >
              Pending
            </button>
            <button
              className="button is-white dropdown-item has-text-grey"
              data-testid="drop-down-item-Closed"
              onMouseDown={[Function]}
            >
              Closed
            </button>
            <button
              className="button is-white dropdown-item has-text-grey"
              data-testid="drop-down-item-Cancelled"
              onMouseDown={[Function]}
            >
              Cancelled
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="is-hidden-tablet column p-0 mt-2 mb-5"
    >
      <div
        aria-controls="dropdown-menu"
        aria-haspopup="true"
        className="dropdown is-right is-flex is-flex-grow-1"
        onBlur={[Function]}
      >
        <div
          className="dropdown-trigger columns m-0 is-flex-grow-1"
        >
          <button
            aria-controls="dropdown-menu"
            aria-haspopup="true"
            className="button rounded-sm is-outlined border-light column m-0 py-0 px-3 is-full-mobile"
            onClick={[Function]}
          >
            <span
              className="has-tooltip-arrow has-tooltip-top is-flex is-flex-grow-1"
              data-tooltip="Filter proposals based on status"
            >
              <div
                className="is-flex is-flex-grow-1 is-align-items-center is-justify-content-space-between has-text-grey small-text"
              >
                All
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-caret-down has-text-black"
                  data-icon="caret-down"
                  data-prefix="fas"
                  focusable="false"
                  height="15"
                  role="img"
                  viewBox="0 0 320 512"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M310.6 246.6l-127.1 128C176.4 380.9 168.2 384 160 384s-16.38-3.125-22.63-9.375l-127.1-128C.2244 237.5-2.516 223.7 2.438 211.8S19.07 192 32 192h255.1c12.94 0 24.62 7.781 29.58 19.75S319.8 237.5 310.6 246.6z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </span>
          </button>
        </div>
        <div
          className="dropdown-menu column p-0 is-full-mobile"
          id="dropdown-menu"
          role="menu"
        >
          <div
            className="dropdown-content"
          >
            <button
              className="button is-white dropdown-item has-text-grey is-active"
              data-testid="drop-down-item-All"
              onMouseDown={[Function]}
            >
              All
            </button>
            <button
              className="button is-white dropdown-item has-text-grey"
              data-testid="drop-down-item-Active"
              onMouseDown={[Function]}
            >
              Active
            </button>
            <button
              className="button is-white dropdown-item has-text-grey"
              data-testid="drop-down-item-Pending"
              onMouseDown={[Function]}
            >
              Pending
            </button>
            <button
              className="button is-white dropdown-item has-text-grey"
              data-testid="drop-down-item-Closed"
              onMouseDown={[Function]}
            >
              Closed
            </button>
            <button
              className="button is-white dropdown-item has-text-grey"
              data-testid="drop-down-item-Cancelled"
              onMouseDown={[Function]}
            >
              Cancelled
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="is-flex is-flex-direction-column"
  />,
]
`;

exports[`ProposalList UI Component renders correctly with no params 1`] = `
Array [
  <div
    className="columns m-0"
  >
    <div
      className="column p-0 is-10-tablet is-hidden-mobile"
    >
      <h1
        className="title is-4 mb-6"
        style={
          Object {
            "lineHeight": "1.67",
          }
        }
      >
        Proposals
      </h1>
    </div>
    <div
      className="is-hidden-mobile column p-0"
    >
      <div
        aria-controls="dropdown-menu"
        aria-haspopup="true"
        className="dropdown is-right is-flex is-flex-grow-1"
        onBlur={[Function]}
      >
        <div
          className="dropdown-trigger columns m-0 is-flex-grow-1"
        >
          <button
            aria-controls="dropdown-menu"
            aria-haspopup="true"
            className="button rounded-sm is-outlined border-light column m-0 py-0 px-3 is-full-mobile"
            onClick={[Function]}
          >
            <span
              className="has-tooltip-arrow has-tooltip-top is-flex is-flex-grow-1"
              data-tooltip="Filter proposals based on status"
            >
              <div
                className="is-flex is-flex-grow-1 is-align-items-center is-justify-content-space-between has-text-grey small-text"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-caret-down has-text-black"
                  data-icon="caret-down"
                  data-prefix="fas"
                  focusable="false"
                  height="15"
                  role="img"
                  viewBox="0 0 320 512"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M310.6 246.6l-127.1 128C176.4 380.9 168.2 384 160 384s-16.38-3.125-22.63-9.375l-127.1-128C.2244 237.5-2.516 223.7 2.438 211.8S19.07 192 32 192h255.1c12.94 0 24.62 7.781 29.58 19.75S319.8 237.5 310.6 246.6z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </span>
          </button>
        </div>
        <div
          className="dropdown-menu column p-0 is-full-mobile"
          id="dropdown-menu"
          role="menu"
        >
          <div
            className="dropdown-content"
          />
        </div>
      </div>
    </div>
    <div
      className="is-hidden-tablet column p-0 mt-2 mb-5"
    >
      <div
        aria-controls="dropdown-menu"
        aria-haspopup="true"
        className="dropdown is-right is-flex is-flex-grow-1"
        onBlur={[Function]}
      >
        <div
          className="dropdown-trigger columns m-0 is-flex-grow-1"
        >
          <button
            aria-controls="dropdown-menu"
            aria-haspopup="true"
            className="button rounded-sm is-outlined border-light column m-0 py-0 px-3 is-full-mobile"
            onClick={[Function]}
          >
            <span
              className="has-tooltip-arrow has-tooltip-top is-flex is-flex-grow-1"
              data-tooltip="Filter proposals based on status"
            >
              <div
                className="is-flex is-flex-grow-1 is-align-items-center is-justify-content-space-between has-text-grey small-text"
              >
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-caret-down has-text-black"
                  data-icon="caret-down"
                  data-prefix="fas"
                  focusable="false"
                  height="15"
                  role="img"
                  viewBox="0 0 320 512"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M310.6 246.6l-127.1 128C176.4 380.9 168.2 384 160 384s-16.38-3.125-22.63-9.375l-127.1-128C.2244 237.5-2.516 223.7 2.438 211.8S19.07 192 32 192h255.1c12.94 0 24.62 7.781 29.58 19.75S319.8 237.5 310.6 246.6z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </span>
          </button>
        </div>
        <div
          className="dropdown-menu column p-0 is-full-mobile"
          id="dropdown-menu"
          role="menu"
        >
          <div
            className="dropdown-content"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="is-flex is-flex-direction-column"
  />,
]
`;

exports[`ProposalList UI Component renders correctly with regular params 1`] = `
Array [
  <div
    className="columns m-0"
  >
    <div
      className="column p-0 is-10-tablet is-hidden-mobile"
    >
      <h1
        className="title is-4 mb-6"
        style={
          Object {
            "lineHeight": "1.67",
          }
        }
      >
        Proposals
      </h1>
    </div>
    <div
      className="is-hidden-mobile column p-0"
    >
      <div
        aria-controls="dropdown-menu"
        aria-haspopup="true"
        className="dropdown is-right is-flex is-flex-grow-1"
        onBlur={[Function]}
      >
        <div
          className="dropdown-trigger columns m-0 is-flex-grow-1"
        >
          <button
            aria-controls="dropdown-menu"
            aria-haspopup="true"
            className="button rounded-sm is-outlined border-light column m-0 py-0 px-3 is-full-mobile"
            onClick={[Function]}
          >
            <span
              className="has-tooltip-arrow has-tooltip-top is-flex is-flex-grow-1"
              data-tooltip="Filter proposals based on status"
            >
              <div
                className="is-flex is-flex-grow-1 is-align-items-center is-justify-content-space-between has-text-grey small-text"
              >
                All
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-caret-down has-text-black"
                  data-icon="caret-down"
                  data-prefix="fas"
                  focusable="false"
                  height="15"
                  role="img"
                  viewBox="0 0 320 512"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M310.6 246.6l-127.1 128C176.4 380.9 168.2 384 160 384s-16.38-3.125-22.63-9.375l-127.1-128C.2244 237.5-2.516 223.7 2.438 211.8S19.07 192 32 192h255.1c12.94 0 24.62 7.781 29.58 19.75S319.8 237.5 310.6 246.6z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </span>
          </button>
        </div>
        <div
          className="dropdown-menu column p-0 is-full-mobile"
          id="dropdown-menu"
          role="menu"
        >
          <div
            className="dropdown-content"
          >
            <button
              className="button is-white dropdown-item has-text-grey is-active"
              data-testid="drop-down-item-All"
              onMouseDown={[Function]}
            >
              All
            </button>
            <button
              className="button is-white dropdown-item has-text-grey"
              data-testid="drop-down-item-Active"
              onMouseDown={[Function]}
            >
              Active
            </button>
            <button
              className="button is-white dropdown-item has-text-grey"
              data-testid="drop-down-item-Pending"
              onMouseDown={[Function]}
            >
              Pending
            </button>
            <button
              className="button is-white dropdown-item has-text-grey"
              data-testid="drop-down-item-Closed"
              onMouseDown={[Function]}
            >
              Closed
            </button>
            <button
              className="button is-white dropdown-item has-text-grey"
              data-testid="drop-down-item-Cancelled"
              onMouseDown={[Function]}
            >
              Cancelled
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="is-hidden-tablet column p-0 mt-2 mb-5"
    >
      <div
        aria-controls="dropdown-menu"
        aria-haspopup="true"
        className="dropdown is-right is-flex is-flex-grow-1"
        onBlur={[Function]}
      >
        <div
          className="dropdown-trigger columns m-0 is-flex-grow-1"
        >
          <button
            aria-controls="dropdown-menu"
            aria-haspopup="true"
            className="button rounded-sm is-outlined border-light column m-0 py-0 px-3 is-full-mobile"
            onClick={[Function]}
          >
            <span
              className="has-tooltip-arrow has-tooltip-top is-flex is-flex-grow-1"
              data-tooltip="Filter proposals based on status"
            >
              <div
                className="is-flex is-flex-grow-1 is-align-items-center is-justify-content-space-between has-text-grey small-text"
              >
                All
                <svg
                  aria-hidden="true"
                  className="svg-inline--fa fa-caret-down has-text-black"
                  data-icon="caret-down"
                  data-prefix="fas"
                  focusable="false"
                  height="15"
                  role="img"
                  viewBox="0 0 320 512"
                  width="30"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M310.6 246.6l-127.1 128C176.4 380.9 168.2 384 160 384s-16.38-3.125-22.63-9.375l-127.1-128C.2244 237.5-2.516 223.7 2.438 211.8S19.07 192 32 192h255.1c12.94 0 24.62 7.781 29.58 19.75S319.8 237.5 310.6 246.6z"
                    fill="currentColor"
                  />
                </svg>
              </div>
            </span>
          </button>
        </div>
        <div
          className="dropdown-menu column p-0 is-full-mobile"
          id="dropdown-menu"
          role="menu"
        >
          <div
            className="dropdown-content"
          >
            <button
              className="button is-white dropdown-item has-text-grey is-active"
              data-testid="drop-down-item-All"
              onMouseDown={[Function]}
            >
              All
            </button>
            <button
              className="button is-white dropdown-item has-text-grey"
              data-testid="drop-down-item-Active"
              onMouseDown={[Function]}
            >
              Active
            </button>
            <button
              className="button is-white dropdown-item has-text-grey"
              data-testid="drop-down-item-Pending"
              onMouseDown={[Function]}
            >
              Pending
            </button>
            <button
              className="button is-white dropdown-item has-text-grey"
              data-testid="drop-down-item-Closed"
              onMouseDown={[Function]}
            >
              Closed
            </button>
            <button
              className="button is-white dropdown-item has-text-grey"
              data-testid="drop-down-item-Cancelled"
              onMouseDown={[Function]}
            >
              Cancelled
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="is-flex is-flex-direction-column"
  >
    <a
      data-testid="proposal-card"
      href="/proposal/1"
      onClick={[Function]}
    >
      <div
        className="border-light rounded-sm mb-5 proposal-card transition-all"
      >
        <div
          className="is-flex proposal-header-spacing "
        >
          <div
            className="columns flex-1 is-mobile has-text-grey m-0 p-0"
          >
            <div
              className="column is-narrow is-flex is-align-items-center pl-0"
            >
              <svg
                fill="none"
                height="15"
                viewBox="0 0 14 15"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  Active
                </title>
                <circle
                  cx="7"
                  cy="7.5"
                  fill="#F4AF4A"
                  opacity="0.5"
                  r="7"
                />
                <circle
                  cx="7"
                  cy="7.5"
                  fill="#F4AF4A"
                  r="3"
                />
              </svg>
            </div>
            <div
              className="column pl-1 pr-1 is-flex is-align-items-center"
              style={
                Object {
                  "flexBasis": "content",
                }
              }
            >
              <code
                className="has-text-grey"
              >
                Active: Ends in 1 days
              </code>
            </div>
            <div
              className="column is-flex is-align-items-center is-justify-content-end proposal-status p-0"
            >
              <code
                className="has-text-grey px-0 smaller-text"
              />
            </div>
          </div>
        </div>
        <hr
          className="my-0"
        />
        <div
          className="px-6 py-5 proposal-body-spacing"
        >
          <h4
            className="proposal-title is-4 mt-1 mb-2"
          >
            test proposal
          </h4>
          <p
            className="has-text-grey mb-4 small-text"
          >
            by 
            0xf8d6e0586b0a20c7
          </p>
          <p
            className="has-text-grey small-text"
          >
            test description
          </p>
        </div>
      </div>
    </a>
    <a
      data-testid="proposal-card"
      href="/proposal/2"
      onClick={[Function]}
    >
      <div
        className="border-light rounded-sm mb-5 proposal-card transition-all"
      >
        <div
          className="is-flex proposal-header-spacing "
        >
          <div
            className="columns flex-1 is-mobile has-text-grey m-0 p-0"
          >
            <div
              className="column is-narrow is-flex is-align-items-center pl-0"
            />
            <div
              className="column pl-1 pr-1 is-flex is-align-items-center"
              style={
                Object {
                  "flexBasis": "content",
                }
              }
            >
              <code
                className="has-text-grey"
              />
            </div>
            <div
              className="column is-flex is-align-items-center is-justify-content-end proposal-status p-0"
            >
              <code
                className="has-text-grey px-0 smaller-text"
              >
                <span
                  className="has-text-white rounded-sm px-2 py-1 has-background-grey-light"
                >
                  Pending
                </span>
              </code>
            </div>
          </div>
        </div>
        <hr
          className="my-0"
        />
        <div
          className="px-6 py-5 proposal-body-spacing"
        >
          <h4
            className="proposal-title is-4 mt-1 mb-2"
          >
            test proposal
          </h4>
          <p
            className="has-text-grey mb-4 small-text"
          >
            by 
            0xf8d6e0586b0a20c7
          </p>
          <p
            className="has-text-grey small-text"
          >
            test description
          </p>
        </div>
      </div>
    </a>
    <a
      data-testid="proposal-card"
      href="/proposal/3"
      onClick={[Function]}
    >
      <div
        className="border-light rounded-sm mb-5 proposal-card transition-all"
      >
        <div
          className="is-flex proposal-header-spacing has-background-white-ter rounded-top"
        >
          <div
            className="columns flex-1 is-mobile has-text-grey m-0 p-0"
          >
            <div
              className="column is-narrow is-flex is-align-items-center pl-0"
            />
            <div
              className="column pl-1 pr-1 is-flex is-align-items-center"
              style={
                Object {
                  "flexBasis": "content",
                }
              }
            >
              <code
                className="has-text-grey"
              >
                
              </code>
            </div>
            <div
              className="column is-flex is-align-items-center is-justify-content-end proposal-status p-0"
            >
              <code
                className="has-text-grey px-0 smaller-text"
              >
                <span
                  className="has-text-white rounded-sm px-2 py-1 has-background-grey"
                >
                  Closed
                </span>
              </code>
            </div>
          </div>
        </div>
        <div
          className="px-6 py-5 proposal-body-spacing"
        >
          <h4
            className="proposal-title is-4 mt-1 mb-2"
          >
            test proposal
          </h4>
          <p
            className="has-text-grey mb-4 small-text"
          >
            by 
            0xf8d6e0586b0a20c7
          </p>
          <p
            className="has-text-grey small-text"
          >
            test description
          </p>
        </div>
      </div>
    </a>
    <a
      data-testid="proposal-card"
      href="/proposal/4"
      onClick={[Function]}
    >
      <div
        className="border-light rounded-sm mb-5 proposal-card transition-all"
      >
        <div
          className="is-flex proposal-header-spacing has-background-white-ter rounded-top"
        >
          <div
            className="columns flex-1 is-mobile has-text-grey m-0 p-0"
          >
            <div
              className="column is-narrow is-flex is-align-items-center pl-0"
            />
            <div
              className="column pl-1 pr-1 is-flex is-align-items-center"
              style={
                Object {
                  "flexBasis": "content",
                }
              }
            >
              <code
                className="has-text-grey"
              />
            </div>
            <div
              className="column is-flex is-align-items-center is-justify-content-end proposal-status p-0"
            >
              <code
                className="has-text-grey px-0 smaller-text"
              >
                <span
                  className="has-text-white rounded-sm px-2 py-1 has-background-grey"
                >
                  Cancelled
                </span>
              </code>
            </div>
          </div>
        </div>
        <div
          className="px-6 py-5 proposal-body-spacing"
        >
          <h4
            className="proposal-title is-4 mt-1 mb-2"
          >
            test proposal
          </h4>
          <p
            className="has-text-grey mb-4 small-text"
          >
            by 
            0xf8d6e0586b0a20c7
          </p>
          <p
            className="has-text-grey small-text"
          >
            test description
          </p>
        </div>
      </div>
    </a>
  </div>,
]
`;
